### PR TITLE
Add convenience function for boxed types

### PR DESCRIPTION
**Pull request by svozza**
As mentioned in #24. I've only refactored the tests for `ValidNumber` given the controversial nature of this PR. :wink:

### CODE REVIEW

File: test/index.js [hunk 1]
```diff
@@ -23,6 +23,26 @@ var errorEq = R.curry(function(type, message, error) {
...
+      break;
+    case 'number':
+      boxed = new Number(val);
+      break;
+    default:
+      throw Error('‘testBoxed‘ only accepts primitives; got ' +
+                  typeof val + '.');
+  }
+  eq(type.test(val), result);
+  eq(type.test(boxed), result);
```
Comment (davidchambers) [inline test/index.js]: Would this remove the need for the `switch` statement?

```diff
-  eq(type.test(boxed), result);
+  eq(type.test(Object(val)), result);
```

Comment (svozza) [inline test/index.js]: Oh nice, that's a neat little trick.

Comment (svozza) [inline test/index.js]: :zap:

**Comment by davidchambers** [comment]
This pull request makes two changes:

  - it replaces the verbose—but perhaps more rigorous—`assert.throws`-style assertions with `eq(Type.test(x), false)`-style assertions; and

  - it introduces `testBoxed` to reduce repetition when testing both primitives and their boxed counterparts for membership in a type.

I'd like to see these changes made in separate pull requests. I want to ensure we don't overvalue the second change as a result of seeing it intermingled with the first change. Stefano, would you mind opening a separate pull request for the first change?

**Comment by svozza** [comment]
Only problem with that is if I remove the tests that use `testBoxed`, I get a linting error about defining an unused function.

**Comment by davidchambers** [comment]
How do you feel about this pull request, Stefano? If you're excited about it, please update it. If not, let's close it.

**Comment by svozza** [comment]
I can't say I'm excited about it so I think we can close.

**Pull request closed without merging**